fog/test-client/src/test_client.rs

```diff
@@ -669,33 +684,50 @@ impl TestClient {
     }
 
     /// Conduct an atomic swap transfer between two clients
-    /// Returns the transaction and the block count of the node it was submitted
-    /// to.
     ///
-    /// The target client builds a signed contingent input, then the source
-    /// client incorporates this into a Tx and submits it.
+    /// The source client builds an SCI, and the target client builds the swap
+    /// transaction. This only builds and submits the transactions, it does not
+    /// confirm it.
+    ///
+    /// The source client's balance is expected to go down by result.value2 of
+    /// tok2, and go up by result.value1 of tok1.
+    ///
+    /// The target client's balance is expected to go down by result.value1 +
+    /// result.fee of tok1, and go up by result.value2 of tok2.
+    ///
     /// This only builds and submits the transaction, it does not confirm it.
     ///
     /// Returns:
-    /// * TransferData: The Tx we submitted, the block count at which we
-    ///   submitted it, and the fee paid
+    /// * SwapTransferData: The Tx we submitted, the block count at which we
+    ///   submitted it, the actual transfer amounts, and the fee paid
     fn atomic_swap(
         &self,
         source_client: &mut Client,
         target_client: &mut Client,
         token_id1: TokenId,
         token_id2: TokenId,
-    ) -> Result<TransferData, TestClientError> {
+        is_partial_fill: bool,
+    ) -> Result<SwapTransferData, TestClientError> {
         self.tx_info.clear();
         let target_address = target_client.get_account_key().default_subaddress();
+
+        let mut rng = McRng::default();
+
+        let tok1_val = 1 + thread_rng().gen_range(0, self.policy.transfer_amount);
+        let tok2_val = 1 + thread_rng().gen_range(0, self.policy.transfer_amount);
```

Comment:
Why not use the `rng` declared right before?
```suggestion
        let tok1_val = 1 + rng.gen_range(0, self.policy.transfer_amount);
        let tok2_val = 1 + rng.gen_range(0, self.policy.transfer_amount);
```

Reply:
it doesn't compile, this function requires the `Rng` trait, but `McRng` only implement `RngCore`. This is ultimately because we don't pull `rand` into the enclave, we pull `rand-core`.